tests/request.spec.coffee

```diff
@@ -290,3 +290,64 @@ describe 'Request:', ->
 				.get('body')
 				m.chai.expect(promise).to.eventually.become(result: 'success')
 
+		describe 'given an endpoint that will time out', ->
+
+			NODE_TIMEOUT_ERROR = new Error('node timeout error')
+
+			beforeEach ->
+				@clock = m.sinon.useFakeTimers()
+
+			afterEach ->
+				@clock.restore()
+
+			stubTimeout = -> new Promise (resolve, reject) ->
+				fetchMock.get 'http://infinite-wait.com', (url, opts) ->
+					resolve()
+
+					# Emulate node-fetch timeout behaviour if we use it
+					if not IS_BROWSER and opts.timeout
+						new Promise (resolve, reject) ->
+							setTimeout((-> reject(NODE_TIMEOUT_ERROR)), opts.timeout)
+					# Browser/no timeout fetch() simply never resolves.
+					else new Promise(->)
```

Comment:
Interesting - I do want this to be an empty function. Happy to reformat this if there's a different coffeescript formatting of this that'll make hound happy, or if anybody has a different simple clear way to define a promise that never resolves.

Notably this isn't caught by the linting in the local build. Sounds like hound is using it's own different linting configuration, which doesn't seem like a good thing.